File: problems/0009_palindrome-number/to-string/Solution.py
#!/usr/bin/python3

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        elif x < 10:
            return True
        s = str(x)
        i, j = 0, len(s) - 1
        while i < j:
            if s[i] != s[j]:
                return False
            i += 1
            j -= 1
        return True

File: problems/0009_palindrome-number/to-string/solution.rb
#!/usr/bin/ruby

# @param {Integer} x
# @return {Boolean}
def is_palindrome(x)
    if x < 0
        return false
    elsif x < 10
        return true
    end
    s = x.to_s
    i, j = [0, s.length - 1]
    while i < j
        if s[i] != s[j]
            return false
        end
        i += 1
        j -= 1
    end
    return true
end
